saleor/graphql/channel/dataloaders.py

```diff
@@ -62,3 +64,13 @@ def channels_by_checkout(orders):
             )
 
         return OrderLineByIdLoader(self.context).load_many(keys).then(channel_by_lines)
+
+
+class ChannelWithOrderCountByIdLoader(DataLoader):
+    context_key = "channel_with_order_count_by_id"
+
+    def batch_load(self, keys):
+        channels = (
+            Channel.objects.all().annotate(order_count=Count("orders")).in_bulk(keys)
```

Comment:
We can do it even faster with `Exists`, something like this should work:
```
from django.db.models import Exists, OuterRef
orders = Order.objects.filter(channel=OuterRef("pk"))
channels = Channel.objects.annotate(has_orders=Exists(orders))
```

https://docs.djangoproject.com/en/3.1/ref/models/expressions/#exists-subqueries